packages/apollo-voyager-keycloak/src/schemaDirectives/hasRole.ts

```diff
@@ -26,25 +25,25 @@ export class HasRoleDirective extends SchemaDirectiveVisitor {
     const { resolve = defaultFieldResolver } = field
     const { error, value } = this.validateArgs()
 
+    if (error) {
+      log.error(`Invalid hasRole directive on field ${field.name}`, error)
```

Comment:
Just a note on this error check. It's checking that the `hasRole` args are valid. Previously it was inside the `field.resolve` function. This was to stop the `data-sync-server` crashing at startup time when someone misconfigured a `hasRole`. Instead, an error would be sent back to a client at runtime if they tried to call a query/mutation that had an invalid `hasRole`.

Now that we're in this framework mode, the previous error handling logic is a bad idea. This change ensures an error is thrown as soon as the directive is loaded if there is an issue.